src/coreos.osbuild.mpp.yaml

```diff
@@ -0,0 +1,228 @@
+version: '2'
+mpp-vars:
+  release: 38
+mpp-define-image:
+  id: image
+  #10G
+  size: '10737418240'
+  table:
+    uuid: 00000000-0000-4000-a000-000000000001
+    label: gpt
+    partitions:
+      - id: BIOS-BOOT
+        size: 2048
+        type: 21686148-6449-6E6F-744E-656564454649
+        bootable: true
+        uuid: FAC7F1FB-3E8D-4137-A512-961DE09A5549
+      - id: EFI-SYSTEM
+        size: 260096
+        type: C12A7328-F81F-11D2-BA4B-00A0C93EC93B
+        uuid: 68B2905B-DF3E-4FB3-80FA-49D1E773AA33
+      - id: boot
+        size: 786432
+        type: 0FC63DAF-8483-4772-8E79-3D69D8477DE4
+        uuid: 61B2905B-DF3E-4FB3-80FA-49D1E773AA32
+      - id: root
+        # XXX: Dynamically set this size in the future
+        size: 4194304
+        type: 0FC63DAF-8483-4772-8E79-3D69D8477DE4
+        uuid: CA7D7CCB-63ED-4C53-861C-1742536059CC
+pipelines:
+  - name: image-tree
+    source-epoch: 1659397331
+    stages:
+      - type: org.osbuild.ostree.init-fs
+      - type: org.osbuild.ostree.os-init
+        options:
+          osname: fedora-coreos
+      - type: org.osbuild.ostree.config
+        options:
+          repo: /ostree/repo
+          config:
+            sysroot:
+              readonly: false
+              bootloader: none
+      - type: org.osbuild.mkdir
+        options:
+          paths:
+            - path: /boot/efi
+              mode: 448
+      - type: org.osbuild.ignition
+      - type: org.osbuild.ostree.deploy
+        options:
+          osname: fedora-coreos
+          remote: fedora
+          mounts:
+            - /boot
+            - /boot/efi
+          kernel_opts:
+            - rw
+            - console=tty0
+            - console=ttyS0
+            - ignition.platform.id=qemu
+            - '$ignition_firstboot'
+        inputs:
+          commits:
+            type: org.osbuild.ostree
+            origin: org.osbuild.source
+            mpp-resolve-ostree-commits:
+              commits:
+                - ref: $ref
+                  remote:
+                    url: $repourl
+      - type: org.osbuild.ostree.selinux
+        options:
+          deployment:
+            osname: fedora-coreos
+            ref: ostree/1/1/0
```

Comment:
This bit is actually poking into what I'd call an ostree implementation detail.  Probably another case of "assume exactly one deployment and use its commit digest".

Reply:
yeah, the reason I did this is because for the deploy-via-container case there is no easy `ref:` we could find to use. For what we are doing here, though, we can just use the ref we are already passing in and just use `ref: $ref` here.

I think what you are suggesting, though, is that OSBuild be modified to make specifying an osname/ref optional and just "do the right thing" in the most common case where there is only one deployment.